File: frontend/src/objects/Pet/PetRequest.ts
export interface PetRequest {
    IdPet: number | undefined | null,
    Nome: string | undefined | null,
    FotoBase64: string | undefined | null,
    Raca: string | undefined | null,
    Tamanho: number | undefined | null,
    Cep: string | undefined | null,
    Cidade: string | undefined | null,
    Bairro: string | undefined | null,
    Rua: string | undefined | null,
    Descricao: string | undefined | null,
    TipoCadastro: number | undefined | null,
    UsuarioCadastroId: string | undefined | null
}

File: backend/PetTrackDotnet/Aplication/Models/Request/Pet/PetRequest.cs
using Infraestrutura.Enum;

namespace Aplication.Models.Request.Pet;

public class PetRequest
{
    public int? IdPet { get; set; }
    public string? Nome { get; set; } = null!;
    public string? Descricao { get; set; } = null!;
    public string? FotoBase64 { get; set; } = null!;
    public string? Raca { get; set; } = null!;
    public string? Cep { get; set; } 
    public string? Cidade { get; set; } 
    public string? Rua { get; set; } 
    public string? Bairro { get; set; } 
    public int? UsuarioCadastroId { get; set; }
    public ETamanho Tamanho { get; set; }
    public ETipoCadastro TipoCadastro { get; set; }
}